Validate Object pointers on binder method calls

KoBeWi opened the pull request:
Fixes #80339

Not sure if it's the right fix, probably not.
Passing a freed object now results in a familiar error:
```
Invalid type in function 'add_child' in base 'Node2D (GodotCrasher.gd)'. The Object-derived class of argument 1 (previously freed) is not a subclass of the expected argument class.
```

aaronfranke committed:
Rename VariantZeroAssigner to VariantDefaultInitializer

And the method zero() is now init()

DarkMessiah committed:
Implement project-wide node groups

vnen committed:
Add option in SCons to copy environment variables

This allows custom environment variables to be used during the build,
which is useful when using alternative compilation tools such as caches
and distributed build systems.

Commit:
Fix missing time for some script functions in profiler

Fixes the issue by adding a mechanism by which the functions that were
previously disappearing can be profiled too. This is optional with
an editor setting, since collecting more information naturally slows the engine
further while profiling.

Fixes #23715, #40251, #29049

Calinou committed:
Add a editor FileSystem dock action to open a terminal in selected folder

This is useful to enter some commands without having to open a separate
terminal and `cd` to the project folder.

bruvzg committed:
[Web] Add IME input support.

EddieBreeg committed:
Fix potential integer underflow in rounded up divisions

A new `Math::division_round_up()` function was added, allowing for easy
and correct computation of integer divisions when the result needs to
be rounded up.

Fixes #80358.

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

AThousandShips committed:
Prevent mapping areas with invalid IDs for `Area2D/3D`

This occurs when areas are created directly from the servers, and no
instance is linked.

BlueCube3310 committed:
Add support for loading more DDS formats

aaronfranke committed:
Add export settings to the export dialog for GLTF

Occalepsus committed:
Fix body leaving area gravity influence

bruvzg committed:
[PopupMenu] Handle physical_keycode and key_label of shortcuts.

paulloz committed:
Add tests for source generators

- Bootstrap xUnit project to test source generators
- Implement source generator tests
- Better tests structure (put test data in cs files)
- Enable `ScriptSerializationGeneratorTests`
- Enable `ScriptPathAttributeGeneratorTests`
- Fix `NesterClass` -> `NestedClass`
- Use `Path.Combine` when dealing with paths
- Copy test data to the output directory

KoBeWi committed:
Improve editing of min/max particle properties

SaracenOne committed:
Fix setting bezier track handle mode from inspector.

RandomShaper committed:
BitField: allow chained set_flag(), make constructors constexpr

RandomShaper committed:
RBMap: Add explicit copy operators to iterators

Absence thereof is deprecated and breaks builds on most compilers.

Bonus: Fix parameter naming style throughout.

BastiaanOlij committed:
Merge passes in Vulkan mobile renderer

bruvzg committed:
[Label] Add get_character_bounds method to get bounding rectangles of the characters.

timothyqiu committed:
Make AnimationTree reference AnimationPlayer instead of AnimationMixer

AThousandShips committed:
Prevent encoding/decoding objects that cannot be instantiated

bruvzg committed:
[Windows] Make Windows Ink tablet API default, add dummy driver to disable tablet input.

KoBeWi committed:
Validate Object pointers on binder method calls

AThousandShips commented on core/variant/binder_common.h:
```diff
@@ -224,6 +224,14 @@ struct PtrToArg<char32_t> {
+#ifdef DEBUG_ENABLED
+		bool freed;
+		Object *object = p_variant.get_validated_object_with_check(freed);
+		if (freed) {
```
This will only catch if the variant is an object, and it was freed, not if it's `null` to start with, or not an object

KoBeWi replied:
I think `can_convert_strict()` already ensures that.

HolonProduction committed:
Fix updating delimiter cache of `CodeEdit` when typing on the first line

ajreckof committed:
Rework update_property for array.

Apply suggestions from code review

Co-Authored-By: Tomek <kobewi4e@gmail.com>

miv391 committed:
Add property change guards to Sprite2D and AnimatedSprite2D

miv391 committed:
Fix invalid frame index when Sprite2D's hframes or vframes has been changed

SomeRanDev committed:
Fix PhysicsShapeQueryParameters3D.motion type wrongly set to Vector2

xiongyaohua committed:
unify parameter names for 'RendererCanvasCull'

chocola-mint committed:
Make it possible to show code hint and code completion at the same time

Make code completion position more consistent

Add whitespace before if

KoBeWi committed:
Copy track update mode when adding reset key

TitanNano committed:
GDScriptAnalyzer is unaware of properties from other Languages

Co-authored-by: K. S. Ernest (iFire) Lee <fire@users.noreply.github.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

bs-mwoerner committed:
Changed HTML input color_changed logic.
Made the HTML field send change events based on whether the new color's string is different from the previous color's string (instead of whether the new string parses to the current color value).
Previously, updating the color value even when the corresponding string hadn't changed would unnecessarily quantize the color value to 8 bits just by opening/closing the Color Picker.

Repiteo committed:
Implement function to throw on null pointers

• Specifically: checks if ptr == IntPtr.Zero

TheSofox committed:
Fix so undoing complex operations in TextEdit will restore selections.

TheSofox committed:
Remember last 'color_mode' and 'picker_shape' in 'ColorPicker's in the editor

Maran23 committed:
Return false when no '.editor.ctex' file exists so that the file will be reimported after and no error is printed

ZeferinoI committed:
Update the description of the method get_connection_line in GraphEdit.xml

The form should be like { from_port: 0, from_node: "GraphNode name 0", to_port: 1, to_node: "GraphNode name 1" }

YuriSizov committed:
Merge pull request #76591 from HolonProduction/complete-tween

Autocomplete properties in `tween_property`

YuriSizov committed:
Merge pull request #78206 from alula/node-count-in-group

Add SceneTree.get_node_count_in_group()

YuriSizov committed:
Merge pull request #82980 from Repiteo/c#-godotsharp-nullable-structs

[C#] Enable nullability for variant structs

YuriSizov committed:
Merge pull request #83957 from kitbdev/scene-tab-reuse

Only update `EditorSceneTabs` tabs when necessary

YuriSizov committed:
Merge pull request #84660 from groud/better_tileset_polygons

Move tile transforms handling cache to TileData

YuriSizov committed:
Merge pull request #84754 from aXu-AP/sprite2dplugin-rect

Add region rect and frames support to Sprite2DPlugin converter

YuriSizov committed:
Merge pull request #84837 from synalice/issue-83366-colored-folders

Fix folder color not showing up in file dialogs

YuriSizov committed:
Merge pull request #84882 from Invertex/select-instance-asset

Add "Show In FileSystem" to the Scene hierarchy right-click

YuriSizov committed:
Merge pull request #85071 from Rubonnek/remove-unnecessary-assignments

Remove unnecessary assignments

YuriSizov committed:
Merge pull request #85362 from Flavelius/params_copy_paste

Make copy & paste params skip `resource_path`

YuriSizov committed:
Merge pull request #85387 from jsjtxietian/fix-forward-plus-crash

Use render method from OS instead of project settings in compositor RD

YuriSizov committed:
Merge pull request #85879 from berrybus/deselect-audio-effects

Fix permanently selected audio bus effects

YuriSizov committed:
Merge pull request #86065 from bruvzg/non_vis_no_draw

[TextServer] Do not draw non-visual characters.

YuriSizov committed:
Merge pull request #86066 from Wiltof/tilemapPaint

Reset TileMap editor `drag_type` when the toolbar mode is not selected

KoBeWi committed:
Properly select the newly duplicated file

Repiteo committed:
Replace `GDVIRTUAL_CALL` with `GDVIRTUAL_REQUIRED_CALL` where applicable

YuriSizov committed:
Fix various issues with the folder color lookup

YuriSizov committed:
Merge pull request #81220 from BlueCube3310/dds-formats-extended

Support loading more DDS formats

YuriSizov committed:
Merge pull request #82561 from ershn/fix_navigation_agents_is_target_reached_behavior

Make `target_desired_distance` affect the navigation of `NavigationAgent2D/3D`

YuriSizov committed:
Merge pull request #86129 from TheSofox/color-picker-memory

Remember last `color_mode` and `picker_shape` in `ColorPicker`s in the editor

YuriSizov committed:
Merge pull request #86164 from KoBeWi/stop_select_faking

Properly select the newly duplicated file

YuriSizov committed:
Merge pull request #86171 from YuriSizov/editor-what-colors-what-sounds

Fix various issues with the folder color lookup

KoBeWi committed:
Fix file disappearing when renaming dependencies

YuriSizov committed:
Correctly register editor-only module classes with the API

TheSofox committed:
Fixed duplicating nodes at different depths in SceneTreeDock

Rindbee committed:
Clear monitoring in `Area*` when its space changes to invalid

So that it can work properly when the space changes to valid again.

Change `space` in advance to prevent disabled areas from being queried again.

TheSofox committed:
Fixes Node selection when navigating SceneTree with Arrow keys

clayjohn committed:
Reduce the number of samplers used by the scene shaders

The hard limit for Apple devices is 16 samplers per stage. So we need to ensure we use less than that

YuriSizov committed:
Merge pull request #70770 from vnen/scons-clone-env-vars

Add option in SCons to clone env variables

YuriSizov committed:
Merge pull request #73896 from vnen/object-null-boolean-consistency

Make freed object different than null in comparison operators

YuriSizov committed:
Merge pull request #78095 from aaronfranke/dict-get-or-set-default

Add a `get_or_add` method to Dictionary

YuriSizov committed:
Merge pull request #79362 from bruvzg/web_ime

[Web] Add IME input support

YuriSizov committed:
Merge pull request #81809 from Rindbee/clean-up-maps-in-Area

Clear monitoring in `Area*` when its space changes to invalid

YuriSizov committed:
Merge pull request #82884 from KoBeWi/commit_message_cached,_please_refresh

Clarify some descriptions regarding resource cache

YuriSizov committed:
Merge pull request #83533 from SaracenOne/bezier_edit_mode_fix

Fix setting bezier track handle mode from inspector

YuriSizov committed:
Merge pull request #84169 from BastiaanOlij/vulkan_mobile_merge_passes

Merge passes in Vulkan mobile renderer

YuriSizov committed:
Merge pull request #84583 from timothyqiu/path-to-ap

Make AnimationTree reference AnimationPlayer instead of AnimationMixer

YuriSizov committed:
Merge pull request #85163 from BastiaanOlij/openxr_play_area

OpenXR: Add support for retrieving play area

YuriSizov committed:
Merge pull request #85317 from miv391/fix-breaking-frame-index-in-sprites

Fix invalid `frame` index when Sprite2D's `hframes` or `vframes` has been changed

YuriSizov committed:
Merge pull request #85575 from TokageItLab/Make-unstore-library

Make unstore AnimationLibrary if AnimationTree is assigned AnimationPlayer

YuriSizov committed:
Merge pull request #85603 from eldidou/speed-up-get-must-clear-dependencies

Speed up `GDScript::get_must_clear_dependencies()`

YuriSizov committed:
Merge pull request #85842 from BlueCube3310/rgtc-compression

Improve normal map VRAM Compression with RGTC

YuriSizov committed:
Merge pull request #85967 from BlueCube3310/squish-ra-as-rg-fix

Fix `squish` DXT5 RA-As-RG channel swapping

YuriSizov committed:
Merge pull request #86137 from Maran23/editor-error-metadata

Do not print error when editor meta was not found as it will be reimported anyway after

YuriSizov committed:
Merge pull request #86161 from ZeferinoI/patch-1

Update the description of the method get_connection_list in GraphEdit

YuriSizov committed:
Merge pull request #86177 from KoBeWi/depren_never_forgets

Fix file disappearing when renaming dependencies

YuriSizov committed:
Merge pull request #86209 from YuriSizov/main-fix-editor-only-module-classes

Correctly register editor-only module classes with the API

YuriSizov committed:
Merge pull request #86219 from clayjohn/RD-samplers-hotfix

Reduce the number of samplers used by the scene shaders

BlueCube3310 committed:
Mention etcpak changes in thirdparty/README.md

Faless committed:
[MP] Fix complete_auth notifying the wrong peer

The SceneMultiplayer complete_auth method was not configuring the
multiplayer peer correctly, causing it to potentially send the
notification to the wrong peer, on the wrong channel, and/or with an
incorrect transfer mode.

TitanNano committed:
Include name field in MethodInfo operator ==

aaronfranke committed:
Use `set_value_no_signal` in editor property code

Calinou committed:
Add input action name to window title in input map editor

YeldhamDev committed:
Fix scene dock search losing its default tooltip after typing an invalid filter

bruvzg committed:
[macOS] Fix updating editor tab titles in the dock menu.

Delsin-Yu committed:
Correctly free relevant scripts when closing scene tabs

Kimau committed:
Expose a method to get hovered Control in Viewport

theraot committed:
Expose `Script.get_global_name()`

YuriSizov committed:
Merge pull request #59186 from aaronfranke/vardefinit

Rename `VariantZeroAssigner` to `VariantDefaultInitializer`

YuriSizov committed:
Merge pull request #79460 from AThousandShips/area_object

Prevent mapping areas with invalid IDs for `Area2D/3D`

YuriSizov committed:
Merge pull request #80487 from theraot/expose_set_global_name

Expose `Script.get_global_name()`

YuriSizov committed:
Merge pull request #82668 from mrjustaguy/master

Improve split blending logic for Vulkan

YuriSizov committed:
Merge pull request #83987 from bruvzg/macos_window_and_help

[macOS] Add default Window and Help menus, allow special menu customization.

YuriSizov committed:
Merge pull request #84898 from rsubtil/fix_dap_breakpoints_closed_scripts

Fix DAP breakpoints being cleared on closed scripts

YuriSizov committed:
Merge pull request #85311 from miv391/add-change-guards-to-sprite2d

Add property change guards to Sprite2D, Sprite3D and AnimatedSprite2D

YuriSizov committed:
Merge pull request #85749 from bs-mwoerner/color_picker_quantize_fix

Fix unnecessarily quantizing current color in color picker

YuriSizov committed:
Merge pull request #85966 from Kimau/claire/expose_mouseover

Expose a method to get hovered Control in Viewport

YuriSizov committed:
Merge pull request #85991 from jsjtxietian/fix-bad-pos-state

Fix `FileAccessPack::get_buffer` updating position past the length of file

YuriSizov committed:
Merge pull request #86008 from Delsin-Yu/4.2

Correctly free relevant scripts when closing scene tabs

YuriSizov committed:
Merge pull request #86044 from KoBeWi/can't_add_if_can't_add

Disable Add button when theme item name is empty

YuriSizov committed:
Merge pull request #86059 from alesliehughes/dynamic_bvh_memmove

Improve DynamicBVH code to make it clearer how the stack/heap works

YuriSizov committed:
Merge pull request #86236 from BlueCube3310/etcpak_rgtc-licence

Mention `etcpak` changes in thirdparty/README.md

YuriSizov committed:
Merge pull request #86278 from YeldhamDev/filter_scene_tooltip

Fix scene dock search losing its default tooltip after typing an invalid filter

YuriSizov committed:
Merge pull request #86280 from Calinou/editor-input-map-editor-action-edit-title

Add input action name to window title in input map editor

aaronfranke committed:
Add PackedRealArray as an alias for Vector<real_t>

clayjohn committed:
Use best fit normals for storing screen space normals

MichaelMacha committed:
Remove pointer to deprecated class page, from Skeleton3D

SkeletonIK3D is now deprecated, but Skeleton3D is recommending looking into its use for complicated inverse kinematics. What's worse, the removed line is not perfectly clear for non-developers and people less experienced with the engine, and could still stand to be improved.

RandomShaper committed:
Split RenderingDevice into API-agnostic and RenderingDeviceDriver parts

Credit and thanks to @bruzvg for multiple build fixes, update of 3rd-party items and MinGW support.

Co-authored-by: bruvzg <7645683+bruvzg@users.noreply.github.com>

YuriSizov committed:
Merge pull request #79785 from KoBeWi/ioioioio

Load project metadata file only when needed

YuriSizov committed:
Merge pull request #81243 from Sauermann/fix-window-events

Fix internal events not being delivered to some Window types

YuriSizov committed:
Merge pull request #84575 from rsburke4/fix-cpu-tangent-force

Fix `CPUParticles2D` tangential force does not match `GPUParticles2D`

YuriSizov committed:
Merge pull request #84652 from timothyqiu/enum-pun

Avoid punning enum names in documentation

YuriSizov committed:
Merge pull request #84708 from bruvzg/winink_def

[Windows] Make Windows Ink tablet API default, add dummy driver to disable tablet input.

YuriSizov committed:
Merge pull request #84826 from chocola-mint/master

Add unit tests for Camera3D

YuriSizov committed:
Merge pull request #84845 from DarkMessiah/tree-disabled-checkbox

Implement disabled state for Checkbox in Tree

YuriSizov committed:
Merge pull request #84940 from Calinou/editor-textureprogressbar-inspector-hide-properties

Hide ineffective TextureProgressBar properties in the editor

YuriSizov committed:
Merge pull request #85000 from bruvzg/ed_ui_direction

[Editor] Add option to override editor UI layout direction.

YuriSizov committed:
Merge pull request #85219 from dsnopek/light-bake-mode-no-gi-method

Remove GI methods in parentheses from light baking options

YuriSizov committed:
Merge pull request #85253 from sandygk/fix-reachable-polygon-closest-to-end-point

[Navigation] Do not use travel cost for minimum when re-selecting end point

YuriSizov committed:
Merge pull request #85447 from smix8/gridmap_navlayer_leftover

Remove GridMap navigation_layers leftover

YuriSizov committed:
Merge pull request #85452 from BleedingXiko/tilemap-typo

Fix typo in TileMap editor tooltip

YuriSizov committed:
Merge pull request #85464 from KoBeWi/only_say_verbose_when_not_verbose

Improve message about resources in use at exit

YuriSizov committed:
Merge pull request #85473 from miv391/fix-horizontal-mouse-wheeling-in-2d-view

Fix horizontal scroll in editor's 2D view

YuriSizov committed:
Merge pull request #85487 from vnen/gdscript-static-register-annotations

GDScript: Make annotations register statically in parser

YuriSizov committed:
Merge pull request #85527 from fdstevex/master

Fix for Cmd-B shortcut conflict on macOS

YuriSizov committed:
Merge pull request #85703 from TitanNano/jovan/gdscript_foreign_script_properties

Make GDScriptAnalyzer aware of properties from other languages

YuriSizov committed:
Merge pull request #85975 from Repiteo/c#-null-ptr-exception

C#: Implement function to throw on null pointers

YuriSizov committed:
Merge pull request #86266 from aaronfranke/editor-prop-no-setting

Use `set_value_no_signal` in editor property code

YuriSizov committed:
Merge pull request #86290 from bruvzg/mac_tab_menu

[macOS] Fix updating editor tab titles in the dock menu.

Calinou committed:
Improve RichTextLabel `install_effect()` documentation

LunaticInAHat committed:
Support unspecified linear size in DDS files

Not all exporters choose to populate that (optional) header field.

nickyfoo committed:
Add getters to navigation servers

Add virtual functions and bind to navigation servers
Implement getters
Add documentation

YuriSizov committed:
Merge pull request #75489 from bruvzg/init_pos_usable

[DisplayServer] Use screen "usable rect" instead of full rect to calculate initial window rect.

YuriSizov committed:
Merge pull request #75623 from reach-satori/profiler_for_builtins

Fix missing script time for some functions in profiler

YuriSizov committed:
Merge pull request #82872 from jsjtxietian/require-editor-restart-when-csg-gizmo-color-changed

Prompt require editor restart to user when gizmo color changed

YuriSizov committed:
Merge pull request #82955 from paulloz/testing-source-generators

Add unit tests for C# source generators

YuriSizov committed:
Merge pull request #82983 from Repiteo/c#-godotsharp-nullable-classes

[C#] Enable nullability for variant classes

YuriSizov committed:
Merge pull request #83785 from Geometror/graphedit-dotted-grid

Add dotted grid to GraphEdit

YuriSizov committed:
Merge pull request #84185 from bruvzg/lbl_char_rects

[Label] Add get_character_bounds method to get bounding rectangles of the characters.

YuriSizov committed:
Merge pull request #84611 from AThousandShips/encode_invalid

Prevent encoding/decoding objects that cannot be instantiated

YuriSizov committed:
Merge pull request #84729 from nickyfoo/add-navigationserver-getters

Add missing getters to navigation servers

YuriSizov committed:
Merge pull request #85436 from chocola-mint/fix-#85401

Make it possible to show code hint and code completion at the same time

YuriSizov committed:
Merge pull request #85470 from KoBeWi/don't_get_child_but_stash_child

Improve Control hiding in PluginConfigDialog

YuriSizov committed:
Merge pull request #85594 from addmix/camera3d-current-bug

Prevent `Camera3D.current` from being set to `true` automatically in the editor.

YuriSizov committed:
Merge pull request #85652 from rsubtil/bugfix-can_grab_focus

Ensure slider grabs focus only when it can

YuriSizov committed:
Merge pull request #86091 from dalexeev/gds-fix-pot-gen-skips-some-nodes

GDScript: Fix POT generator skips some nodes

YuriSizov committed:
Merge pull request #86331 from Calinou/doc-richtexteffect

Improve RichTextLabel `install_effect()` documentation

brno32 committed:
Add note that a large value for outline_size is not recommended

RandomShaper committed:
Make PagedAllocator more compatible (esp., with HashMap)

YuriSizov committed:
Optimize scanning routines in the project manager

YuriSizov committed:
Correctly enforce minimum window size in editor

Also set the maximum size for the language
picked in the project manager.

YuriSizov committed:
Merge pull request #60965 from DarkMessiah/global-groups-implementation

Implement project-wide node groups

YuriSizov committed:
Merge pull request #81260 from KoBeWi/min_maxing_particle_editing

Improve editing of min/max particle properties

YuriSizov committed:
Merge pull request #83623 from rarysson/label-progressbar-allow-greater

Make ProgressBar reflect the real value in the percent label when "allow greater" is checked

YuriSizov committed:
Merge pull request #84734 from Anutrix/node-doc-11112023

Add description to `AnimationNodeStateMachine.replace_node`

YuriSizov committed:
Merge pull request #84979 from Calinou/scons-web-platform-alias

Alias `platform=javascript` SCons option to `platform=web`

YuriSizov committed:
Merge pull request #85191 from HolonProduction/off-by-one-code-edit

Fix updating delimiter cache of `CodeEdit` when typing on the first line

YuriSizov committed:
Merge pull request #85887 from YuriSizov/editor-enforce-min-window

Correctly enforce minimum window size in editor

YuriSizov committed:
Merge pull request #86118 from TheSofox/complex-undo-select-fix

Fix so undoing complex operations in `TextEdit` will restore selections

YuriSizov committed:
Merge pull request #86169 from Repiteo/gdvirtual-call-refactoring

Replace `GDVIRTUAL_CALL` with `GDVIRTUAL_REQUIRED_CALL` where applicable

YuriSizov committed:
Merge pull request #86218 from TheSofox/arrow-key-scene-tree-nav

Fix `Node` selection when navigating `Tree` with Arrow keys

YuriSizov committed:
Merge pull request #86257 from Faless/mp/fix_complete_auth

[MP] Fix `complete_auth` notifying the wrong peer

YuriSizov committed:
Merge pull request #86271 from YuriSizov/pm-optimize-multiple-ways-of-scanning

Optimize scanning routines in the project manager

YuriSizov committed:
Merge pull request #86324 from aaronfranke/really-packed

Add PackedRealArray as an alias for Vector<real_t>

YuriSizov committed:
Merge pull request #86326 from MichaelMacha/patch-1

Remove pointer to deprecated class page from Skeleton3D

YuriSizov committed:
Merge pull request #86336 from LunaticInAHat/dds_implicit_linearsize

Support unspecified linear size in DDS files

YuriSizov committed:
Merge pull request #83452 from RandomShaper/rd_common

Split `RenderingDevice` into API-agnostic and `RenderingDeviceDriver` parts

raulsntos committed:
C#: Use string literal instead of nameof

Using `nameof` expressions with method parameters requires C# 11 and we're currently on C# 10.

YuriSizov committed:
Merge pull request #86373 from raulsntos/dotnet/fix-nameof

C#: Use string literal instead of nameof

timothyqiu committed:
Emit slider's drag_started signal before the first value change

jsjtxietian committed:
Fix `UndoRedo history mismatch` when creating a new tile atlas

timothyqiu committed:
Clear sub-resources list when no sub-resource exists

warriormaster12 committed:
Fix wrong error message when graphics pipeline creation fails

Calinou committed:
Suggest `--headless` CLI argument in DisplayServer not found error message

This is helpful for newcomers setting up Godot on CI, as this provides
automatic guidance as for why it doesn't run out of the box if
there's no X11 server available.

timothyqiu committed:
Improve action name for ungroup button in Scene dock

jsjtxietian committed:
Add RD_ENABLED when VULKAN_ENABLED or D3D12_ENABLED is added

YuriSizov committed:
Merge pull request #82961 from Occalepsus/fix_body_area_gravity

Fix body leaving area gravity influence

YuriSizov committed:
Merge pull request #83655 from golfinq/small-capsule-shape-fix

Replace confusing code in GodotCapsuleShape2D::get_supports

YuriSizov committed:
Merge pull request #85923 from Calinou/editor-filesystem-dock-add-open-in-terminal

Add a editor FileSystem dock action to open a terminal in selected folder

YuriSizov committed:
Merge pull request #86316 from clayjohn/RD-BFN-normals

Use best fit normals for storing screen space normals

YuriSizov committed:
Merge pull request #86345 from brno32/outline-size-docs

Add note that a large value for `Label.outline_size` is not recommended

YuriSizov committed:
Merge pull request #86377 from timothyqiu/slider-drag

Emit slider's `drag_started` signal before the first value change

YuriSizov committed:
Merge pull request #86387 from jsjtxietian/fix-tile-set

Fix `UndoRedo history mismatch` when creating a new tile atlas

YuriSizov committed:
Merge pull request #86388 from timothyqiu/unique-reset

Clear sub-resources list when no sub-resource exists

YuriSizov committed:
Merge pull request #86405 from warriormaster12/pipeline-message-fix

Fix wrong error message when graphics pipeline creation fails

YuriSizov committed:
Merge pull request #86409 from Calinou/displayserver-failure-message-suggest-headless

Suggest `--headless` CLI argument in DisplayServer not found error message

YuriSizov committed:
Merge pull request #86431 from timothyqiu/ungroup-action

Improve action name for ungroup button in Scene dock

lyuma committed:
gltf: fix three bugs which prevented extracted textures from being refreshed.

1. Extracted texture paths in `GLTFDocument::_parse_image_save_image` at the project root started with res:/// which broke cache invalidation
2. md5 hashes were not being written to generator_parameters for new imports, which led Godot to think the file was manually created.
3. `EditorFileSystem::reimport_append` must emit the `resources_reimported` signal in order for the resource cache to be updated.

Sauermann committed:
Fix D&D viewport position calculation

It is necessary to transform screen position to viewport position.

touilleMan committed:
Fix `ScriptLanguageExtension::_find_function` documentation

RandomShaper committed:
RenderingDevice: Fix uniform sets wrongly assumed to be bound

sepTN committed:
Fix various typos in documentation

bruvzg committed:
[D3D12] Dynamically load Agility SDK.

sepTN committed:
Fix `OS.get_system_font_path` and `OS.get_system_font_path_for_text` to return correct value

RandomShaper committed:
Remove traces of the extinct RID preallocate feature

MJacred committed:
Fix AudioServer::start_playback_stream does not iterate through given p_bus_volumes

rune-scape committed:
Initialize pointers in a_star.cpp

pancelor committed:
project upgrade tool: better space handling in "export"

the regex had two problems; they were not recognizing export statements
that had spaces after "export", or export statements with no space
before "var".

These are both valid syntax, but the conversion was unsuccessful.
This commit fixes both cases.

test case 1:    `export(int)var x = 1`
old conversion: `@export(int)var x = 1`
new conversion: `@export var x: int = 1`

test case 2:    `export (int) var y = 2`
old conversion: `@export (int) var y = 2`
new conversion: `@export var y: int = 2`

TokageItLab committed:
Add cubic_interpolate_in_time_variant() to Animation

nikitalita commented on core/variant/binder_common.h:
```diff
@@ -224,6 +224,14 @@ struct PtrToArg<char32_t> {
+
 		using TStripped = std::remove_pointer_t<T>;
 		if constexpr (std::is_base_of<Object, TStripped>::value) {
 			Object *obj = p_variant;
```
change the line just below this to `return !obj || Object::cast_to<TStripped>(p_variant);`

nikitalita commented:
Just tested this; This PR fixes #86602  (and possibly https://github.com/godotengine/godot/issues/86427 as well). 

Pursuant to https://github.com/godotengine/godot/pull/57205#pullrequestreview-1798728003, recommend changing the line mentioned above.

KoBeWi commented:
This PR seems to break `is_instance_valid()` so it's likely not proper fix after all.

Commit:
Fix `linuxbsd_headers/README.md`

TokageItLab committed:
Make default options of int value track refer to RESET animation

LunaCapra committed:
Correct C# syntax in _validate_property example for the Object class

qwofford committed:
corrected tooltip information Roll Influence property of VehicleWheel3D

Mickeon committed:
Link to the "article" in the docs instead of saying "article"

Daylily-Zeleen committed:
Distinguishs between dynamic library not found and can't be opened.

jsjtxietian committed:
Update get_image doc to mention that it will return an empty image with invalid texture

AThousandShips committed:
[GDScript] Add `StringName` support to `@GDScript.len`

akien-mga committed:
Merge pull request #79316 from aaronfranke/gltf-export

Add export settings to the export dialog for GLTF

akien-mga committed:
Merge pull request #80390 from EddieBreeg/issue_80358

Fix potential integer underflow in rounded up divisions

akien-mga committed:
Merge pull request #80706 from ajreckof/rework-array-update-property

Rework `update_property` for array

akien-mga committed:
Merge pull request #84151 from YeldhamDev/that_was_not_a_resize_you_dunce

Fix `NOTIFICATION_WM_SIZE_CHANGED` firing if the size hasn't changed

akien-mga committed:
Merge pull request #85393 from SomeRanDev/master

Fix PhysicsShapeQueryParameters3D.motion type wrongly set to Vector2

akien-mga committed:
Merge pull request #85421 from xiongyaohua/fix_parameter_names

Unify parameter names for 'RendererCanvasCull'

akien-mga committed:
Merge pull request #86211 from TheSofox/node-duplicate-undo-fix

Fix duplicating multiple nodes at different depths in `SceneTreeDock`

akien-mga committed:
Merge pull request #86259 from TitanNano/jovan/fix_MethodInfo_eq

Include `name` field in MethodInfo operator ==

akien-mga committed:
Merge pull request #86435 from jsjtxietian/fix-fp-unsupport

Add `RD_ENABLED` when `VULKAN_ENABLED` or `D3D12_ENABLED` is added

akien-mga committed:
Merge pull request #86704 from AThousandShips/len_fix

[GDScript] Add `StringName` support to `@GDScript.len`

AThousandShips commented:
Seems the branch is a bit messed up

KoBeWi committed:
Handle built-in shaders when closing scene

akien-mga committed:
Merge pull request #82264 from dalexeev/core-builtin-methods-as-callables

Core: Allow methods of built-in `Variant` types to be used as Callables

akien-mga committed:
Merge pull request #82897 from bruvzg/mac_menu_phy_lbl

[PopupMenu] Handle `physical_keycode` and `key_label` of shortcuts.

akien-mga committed:
Merge pull request #85540 from KoBeWi/update_reset

Copy track update mode when adding reset key

akien-mga committed:
Merge pull request #86088 from dalexeev/gds-fix-static-func-as-callable-in-static-context

GDScript: Fix accessing static function as `Callable` in static context

akien-mga committed:
Merge pull request #86504 from lyuma/gltf_reimport_textures_merry_xmas

gltf: Fix three bugs which prevented extracted textures from being refreshed

akien-mga committed:
Merge pull request #86511 from Sauermann/fix-drag-transform

Fix D&D viewport position calculation

akien-mga committed:
Merge pull request #86520 from touilleMan/fix-ScriptLanguageExtension-_find_function-documentation

Fix `ScriptLanguageExtension::_find_function` documentation

akien-mga committed:
Merge pull request #86549 from sepTN/fix-docs-typos2

Fix various typos in documentation

akien-mga committed:
Merge pull request #86551 from bruvzg/agility_dynamic_load

[D3D12] Dynamically load Agility SDK.

akien-mga committed:
Merge pull request #86552 from sepTN/fix-get_system_font_path

Fix `OS.get_system_font_path` and `OS.get_system_font_path_for_text` to return correct slashes

akien-mga committed:
Merge pull request #86570 from RandomShaper/remove_traces_of_rid_prealloc

Remove traces of the extinct RID preallocate feature

akien-mga committed:
Merge pull request #86584 from MJacred/fix/audio_server_set_playbyck_stream

Fix `AudioServer::start_playback_stream` does not iterate through given `p_bus_volumes`

akien-mga committed:
Merge pull request #86590 from rune-scape/rune-uninitialized-astar

Initialize pointers in a_star.cpp

akien-mga committed:
Merge pull request #86598 from pancelor/master

Project upgrade tool: Better space handling in "export"

akien-mga committed:
Merge pull request #86601 from TokageItLab/add-cubic-interpolation-variant

Add `cubic_interpolate_in_time_variant()` to Animation

akien-mga committed:
Merge pull request #86607 from programneer/linuxbsd-readme-fix

Fix README from `linuxbsd_headers`

akien-mga committed:
Merge pull request #86630 from TokageItLab/value-track-default-option

Make default options of int value track refer to RESET animation

akien-mga committed:
Merge pull request #86649 from SGiygas/patch-1

Correct C# syntax in `_validate_property` example for the Object class

akien-mga committed:
Merge pull request #86666 from Mickeon/merge-this-before-2024-i-bet-you-wont

Link to mentioned articles in the docs instead of just naming them

akien-mga committed:
Merge pull request #86672 from qwofford/qwofford/issue85805

Fix incorrect VehicleWheel3D Roll Influence description

akien-mga committed:
Merge pull request #86682 from Daylily-Zeleen/daylily-zeleen/distinguish_between_dynamic_libaray_not_found_and_can't_open

Distinguish between dynamic library not found and can't be opened.

akien-mga committed:
Merge pull request #86685 from jsjtxietian/update-get-image-doc

Update `get_image` doc to mention that it will return an empty image with invalid texture

akien-mga committed:
Merge pull request #86710 from KoBeWi/do_not_the_shaders

Handle built-in shaders when closing scene

akien-mga committed:
Merge pull request #86522 from RandomShaper/fix_uset_rebind

RenderingDevice: Fix uniform sets wrongly assumed to be bound

KoBeWi commented:
I'm unable to fix it for some reason. Force pushes don't appear. In fact, no push appears in the history, so it's like it broke for no reason 🤔 (I did update the branch though, few days ago. The checks were processing today, so I doubt they were started by that push)

akien-mga commented:
I'm also the one who got the CI failure notification, saying that this PR was "synchronize by akien-mga".

```
odules/gdscript/tests/gdscript_test_runner.cpp:216: ERROR: CHECK( result.passed ) is NOT correct!
  values: CHECK( false )
  logged: /home/runner/work/godot/godot/modules/gdscript/tests/scripts/runtime/features/getter_with_freed_object.gd
          GDTEST_RUNTIME_ERROR
>> SCRIPT ERROR
>> on function: test()
>> runtime/features/getter_with_freed_object.gd
>> 14
>> Invalid type in utility function "is_instance_valid()". Cannot convert argument 1 from Object to Nil.
```

Might be worth redoing the PR if it's misbehaving. Maybe GitHub's support for emoji branch names isn't as good as you'd like ;)
